src/Symfony/Component/Yaml/Inline.php

```diff
@@ -712,15 +721,20 @@ private static function evaluateScalar(string $scalar, int $flags, array &$refer
                             return $time;
                         }
 
-                        try {
-                            if (false !== $scalar = $time->getTimestamp()) {
-                                return $scalar;
+                        $length = \strlen(rtrim($time->format('u'), '0'));
+                        if (0 === $length) {
+                            try {
+                                if (false !== $scalar = $time->getTimestamp()) {
```

Comment:
Do you know why there is 2 ways of generating a timestamp with a possible `false` or `TypeError`?

Reply:
No. It's in the existing code. There's no comment explaining the reason, but tests pass without it.

Reply:
Found it. [2353f19cc2c99c34cd1f126bd95f042cbe458824](https://github.com/symfony/symfony/commit/2353f19cc2c99c34cd1f126bd95f042cbe458824)